compiler/pragmas.nim

```diff
@@ -321,8 +321,12 @@ proc processNote(c: PContext, n: PNode) =
 
     let x = c.semConstBoolExpr(c, n[1])
     n.sons[1] = x
-    if x.kind == nkIntLit and x.intVal != 0: incl(c.config.notes, nk)
-    else: excl(c.config.notes, nk)
+    if x.kind == nkIntLit and x.intVal != 0:
+      incl(c.config.notes, nk)
```

Comment:
This line seems to be the actual bugfix. The other code changes are a refactoring. The refactoring is ok, but is it required for the bugfix?

Reply:
Other code changes are not refactoring. Please see my comment below.